### ISSUE TITLE
Cannot read property 'definitions' of undefined

### ISSUE THREAD

**Issue by aswinckr** [d4rekanguok/gatsby-typescript#31]
I tried to use the `gatsby-plugin-graphql-codegen` on my gatsby site but i'm unable to develop my gatsby site. Am i missing something? Appreciate your help!

On my terminal -
```
 ERROR

Cannot read property 'definitions' of undefined

  TypeError: Cannot read property 'definitions' of undefined

  - index.cjs.js:397 Promise.all.documentFiles.map
    [frontend]/[@graphql-toolkit]/common/index.cjs.js:397:69

  - Array.map

  - index.cjs.js:394 Object.validateGraphQlDocuments
    [frontend]/[@graphql-toolkit]/common/index.cjs.js:394:50
```

`gatsby-config.js` : 

```
{
      resolve: `gatsby-plugin-graphql-codegen`,
},
```

`pages/work.tsx`
```
const Work = ({ data }: any) => (
  <Layout>
    {console.log(data)}
    <SEO title="Work" keywords={[`work`, `portfolio`, `react`]} />
    <h1>This is where I showcase my work</h1>
    <Link to="/about/">Go to page 2</Link>
  </Layout>
);

export const query = graphql`
  query WorksData {
    allSanityWork {
      nodes {
        title
      }
    }
  }
`;

export default Work;
```

**Comment by d4rekanguok** [comment]
Hi @aswinckr, sorry this happens — are you using a gatsby starter or a theme of some sort, if so, which one is it? It would help reproduce this error (a reproduction repo is very, very welcomed.) Thank you for giving this plugin a try!

**Comment by d4rekanguok** [comment]
_Ohh_ have you tried just using the plugin as a string?
```
// gatsby-config.js

module.exports = {
  plugins: [
    'gatsby-plugin-graphql-codegen',
  ]
}
```

**Comment by eXaminator** [comment]
I'm seeing the same error with the 'gatsby-plugin-ts'.

Here is my config (I also tryed it with just the string, same error):

```
{
    resolve: 'gatsby-plugin-ts',
    options: {
        forkTsCheckerPlugin: {
            eslint: true,
        },
        fileName: './src/types/graphql.d.ts',
        typeCheck: process.env.NODE_ENV !== 'production',
    },
},
```

**Comment by d4rekanguok** [comment]
Thanks for taking the time to reporting in, definitely seems like a bug -- sorry about this

**Comment by d4rekanguok** [comment]
The fix published in the following packages:
 - gatsby-plugin-graphql-codegen@2.1.4
 - gatsby-plugin-ts@2.1.4

So it was a weird graphql-toolkit versions mismatched. Migrating to the latest version fixed the issue (see #34). 

Thank you so much @aswinckr and @eXaminator for reporting the issue & allowing me to fix this in a quick manner! Please re-open the issue if the problem persists, hope you have a good time using these plugins.
